heron/uploaders/src/java/com/twitter/heron/uploader/docker/DockerUploader.java

```diff
@@ -0,0 +1,187 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.twitter.heron.uploader.docker;
+
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.uploader.IUploader;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * IUploader Implementation that creates a DockerUploader Image for the Topology
+ * <p>
+ * The created DockerUploader Image will be based on the given base image and include the topology definition
+ * and binary in a fixed location. The uploader optionally pushes the docker image to a remote repository.
+ * Downstream schedulers must take responsibility for leveraging this image, by either building an additional
+ * DockerUploader image based on it with the correct START command
+ * </p>
+ * <p>
+ * The topology definition will be written to "/home/[role]/[topology-name]"
+ * if no role is given "heron" is used
```

Comment:
Is there a way to check if the user didn't supply the role? I don't like the idea of checking for default.